### PR TITLE
Be crud user in organization 66

### PR DESCRIPTION
**Pull request by berki0**

### CODE REVIEW

File: src/main/java/jewellery/inventory/dto/request/UpdateUserPermissionsRequest.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import jakarta.validation.constraints.NotEmpty;
+import jewellery.inventory.model.OrganizationPermission;
+import lombok.Data;
+
+import java.util.List;
+
+@Data
+public class UpdateUserPermissionsRequest {
+  @NotEmpty private List<OrganizationPermission> organizationPermission;
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/dto/request/UpdateUserPermissionsRequest.java]: The list could be empty. We can add a member with no permissions.

File: src/main/java/jewellery/inventory/controller/OrganizationController.java [hunk 2]
```diff
@@ -32,11 +35,46 @@ public OrganizationResponseDto getOrganizationById(@PathVariable UUID id) {
...
-  public OrganizationResponseDto create(@RequestBody @Valid OrganizationRequestDto organizationRequestDto){
+  public OrganizationResponseDto create(
+      @RequestBody @Valid OrganizationRequestDto organizationRequestDto) {
     return organizationService.create(organizationRequestDto);
   }
 
+  @Operation(summary = "Add a user in organization")
+  @ResponseStatus(HttpStatus.CREATED)
+  @PostMapping("/{organizationId}/users")
+  public OrganizationResponseDto addUserInOrganization(
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/controller/OrganizationController.java]: I think it makes more sense to return UserInOrganizationResponseDto here instead of the Organization as there is no information about the user in the OrganizationResponseDto.

File: src/main/java/jewellery/inventory/controller/OrganizationController.java [hunk 3]
```diff
@@ -32,11 +35,46 @@ public OrganizationResponseDto getOrganizationById(@PathVariable UUID id) {
...
+      @PathVariable UUID organizationId,
+      @RequestBody @Valid UserInOrganizationRequestDto userInOrganizationRequestDto) {
+    return organizationService.addUserInOrganization(organizationId, userInOrganizationRequestDto);
+  }
+
+  @Operation(summary = "Delete a user in organization")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  @DeleteMapping("{organizationId}/users/{userId}")
+  public OrganizationResponseDto deleteUserInOrganization(
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/controller/OrganizationController.java]: This could be void

File: src/main/java/jewellery/inventory/controller/OrganizationController.java [hunk 4]
```diff
@@ -32,11 +35,46 @@ public OrganizationResponseDto getOrganizationById(@PathVariable UUID id) {
...
+      @PathVariable UUID organizationId, @PathVariable UUID userId) {
+    return organizationService.deleteUserInOrganization(userId, organizationId);
+  }
+
+  @Operation(summary = "Update a user permissions in organization")
+  @ResponseStatus(HttpStatus.OK)
+  @PutMapping("{organizationId}/users/{userId}")
+  public OrganizationResponseDto updateUserPermissionsInOrganization(
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/controller/OrganizationController.java]: Same here, let's return the UserInOrganization after the update.

File: src/main/java/jewellery/inventory/exception/organization/UserIsNotPartOfOrganizationException.java
Comment (VladoKat) [inline src/main/java/jewellery/inventory/exception/organization/UserIsNotPartOfOrganizationException.java:1]: Should we add this to the GlobalExceptionHandler? I don't see it included in the exception handlers.

File: src/main/java/jewellery/inventory/exception/organization/UserNotHaveUserPermissionException.java
Comment (VladoKat) [inline src/main/java/jewellery/inventory/exception/organization/UserNotHaveUserPermissionException.java:1]: Rename to MissingOrganizationPermissionException as it is more descriptive I think

File: src/main/java/jewellery/inventory/exception/organization/UserNotHaveUserPermissionException.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package jewellery.inventory.exception.organization;
+
+import java.util.UUID;
+
+public class UserNotHaveUserPermissionException extends RuntimeException {
+  public UserNotHaveUserPermissionException(UUID userId, UUID organizationId) {
+    super(
+        "The User with id "
+            + userId
+            + " not have MANAGE_USERS permission for Organization with id "
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/exception/organization/UserNotHaveUserPermissionException.java]: Should we pass the OrganizationPermission enum value as part of the arguments instead of hardcoding it. We will want to reuse this exception in the future

File: src/main/java/jewellery/inventory/mapper/OrganizationMapper.java [hunk 6]
```diff
@@ -26,4 +30,17 @@ public OrganizationResponseDto toResponse(Organization organization) {
...
+    List<UserInOrganizationResponseDto> list = new ArrayList<>();
+    for (int i = 0; i < organization.getUsersInOrganization().size(); i++) {
+      UserInOrganizationResponseDto user = new UserInOrganizationResponseDto();
+      user.setUserId(organization.getUsersInOrganization().get(i).getUser().getId());
+      user.setOrganizationPermissions(
+          organization.getUsersInOrganization().get(i).getOrganizationPermission());
+      list.add(user);
+    }
+    return list;
+  }
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/mapper/OrganizationMapper.java]: Let's either use stream->map or foreach. They would be a bit shorter.

File: src/main/java/jewellery/inventory/model/EventType.java
Comment (VladoKat) [inline src/main/java/jewellery/inventory/model/EventType.java:1]: We should have ORAGNIZATION_USER_UPDATE as well for updating permissions for a user.

File: src/main/java/jewellery/inventory/model/Organization.java [hunk 7]
```diff
@@ -19,21 +19,23 @@
...
+  @Column private String note;
+
   @OneToMany(mappedBy = "organization", cascade = CascadeType.ALL)
   private List<ResourceInOrganization> resourceInOrganization;
+
   @OneToMany(mappedBy = "organization", cascade = CascadeType.ALL)
   private List<Product> productsOwned;
-  @OneToMany(mappedBy = "organization", cascade = CascadeType.ALL)
+
+  @OneToMany(mappedBy = "organization", orphanRemoval = true, cascade = CascadeType.PERSIST)
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/model/Organization.java]: Looks like we have added orphanRemoval and  cascadeType change to CascadeType.PERSIST instead of CascadeType.ALL. Was it causing any problems to be CascadeType.ALL?

File: src/main/java/jewellery/inventory/service/OrganizationService.java [hunk 8]
```diff
@@ -63,4 +118,85 @@ private Organization getOrganization(UUID id) {
         .findById(id)
         .orElseThrow(() -> new OrganizationNotFoundException(id));
   }
+
+  private void validateUserPermission(Organization organization) {
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/service/OrganizationService.java]: We just care about the MANAGE_USERS permission here but still it would be better to add the permission as an argument in this method so that we can reuse it later.

File: src/main/java/jewellery/inventory/service/OrganizationService.java [hunk 9]
```diff
@@ -34,6 +38,57 @@ public OrganizationResponseDto getOrganizationResponse(UUID id) {
...
+    validateUserPermission(organization);
+
+    UserInOrganization userInOrganization =
+        createUserInOrganization(userInOrganizationRequestDto, organization);
+    addUserToOrganization(userInOrganization, organization);
+
+    return organizationMapper.toResponse(organization);
+  }
+
+  @LogCreateEvent(eventType = EventType.ORGANIZATION_USER_DELETE)
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/service/OrganizationService.java]: This should be a LogDeleteEvent?

File: src/main/java/jewellery/inventory/service/OrganizationService.java [hunk 10]
```diff
@@ -34,6 +38,57 @@ public OrganizationResponseDto getOrganizationResponse(UUID id) {
     return organizationMapper.toResponse(getOrganization(id));
   }
 
+  public List<UserInOrganizationResponseDto> getAllUsersInOrganization(UUID organizationId) {
+    Organization organization = getOrganization(organizationId);
+    validateUserInOrganization(organization);
+    return organizationMapper.toUserInOrganizationResponseDtoResponse(organization);
+  }
+
+  public OrganizationResponseDto updateUserPermissionsInOrganization(
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/service/OrganizationService.java]: This should be annotated with LogUpdateEvent?

File: src/main/java/jewellery/inventory/service/OrganizationService.java [hunk 11]
```diff
@@ -63,4 +118,85 @@ private Organization getOrganization(UUID id) {
...
+    organizationRepository.save(organization);
+  }
+
+  private void changeUserPermissionInOrganization(
+      Organization organization,
+      User userForUpdate,
+      List<OrganizationPermission> organizationPermissionList) {
+    organization
+        .getUsersInOrganization()
+        .forEach(
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/service/OrganizationService.java]: Let's not use forEach here. Use somthing like findFirst for the ID and change that specific userInOrganization

File: src/test/java/jewellery/inventory/helper/OrganizationTestHelper.java [hunk 12]
```diff
@@ -25,6 +26,29 @@ public static Organization getTestOrganization() {
     return organization;
   }
 
+  public static Organization getTestOrganizationWithUser(User user) {
+    Organization organization = new Organization();
+    organization.setId(UUID.randomUUID());
+    organization.setName(ORGANIZATION_NAME);
+    organization.setAddress(ORGANIZATION_ADDRESS);
+    organization.setNote(ORGANIZATION_NOTE);
```
Comment (VladoKat) [inline src/test/java/jewellery/inventory/helper/OrganizationTestHelper.java]: This is just copied from the getTestOrganization() method. We can reuse it here like:
```suggestion
    Organization organization = getTestOrganization()
```

File: src/test/java/jewellery/inventory/integration/OrganizationCrudIntegrationTest.java
Comment (VladoKat) [inline src/test/java/jewellery/inventory/integration/OrganizationCrudIntegrationTest.java:1]: Please, create a test for updating user's permissions as well

File: src/test/java/jewellery/inventory/integration/OrganizationCrudIntegrationTest.java [hunk 13]
```diff
@@ -68,6 +87,50 @@ void getOrganizationByIdSuccessfully() {
...
+            HttpMethod.DELETE,
+            null,
+            OrganizationResponseDto.class);
+
+    assertEquals(HttpStatus.NO_CONTENT, response.getStatusCode());
+    assertNull(response.getBody());
+  }
+
+  @Test
+  void AddUserInOrganizationSuccessfully() {
```
Comment (VladoKat) [inline src/test/java/jewellery/inventory/integration/OrganizationCrudIntegrationTest.java]: pls make this test name in lower case

File: src/test/java/jewellery/inventory/unit/service/OrganizationServiceTest.java
Comment (VladoKat) [inline src/test/java/jewellery/inventory/unit/service/OrganizationServiceTest.java:1]: Let's not add successful tests to the unit tests here as they should be covered by the integration ones and therefore these are just duplicates.

File: src/test/java/jewellery/inventory/unit/service/OrganizationServiceTest.java [hunk 14]
```diff
@@ -97,11 +102,103 @@ void createOrganizationSuccessfully() {
...
     UUID fakeId = UUID.fromString("58bda8d1-3b3d-4319-922b-f5bb66623d71");
     assertThrows(
         OrganizationNotFoundException.class,
         () -> organizationService.getOrganizationResponse(fakeId));
   }
+
+  @Test
+  void deleteUserInOrganizationThrowsExceptionWhenNoManageUsersPermission() {
+    when(authService.getCurrentUser()).thenReturn(executorResponseDto);
+    when(userService.getUser(any(UUID.class))).thenReturn(new User());
```
Comment (VladoKat) [inline src/test/java/jewellery/inventory/unit/service/OrganizationServiceTest.java]: We can be a bit more specific here to let the reader know that these are connected.
```suggestion
    when(userService.getUser(executorResponseDto.getId())).thenReturn(new User());
```

**Review by VladoKat — CHANGES_REQUESTED** [review]
Apart from the changes here we should discuss the SystemEvents logging as we haven't discussed what should be saved in the events.

File: src/main/java/jewellery/inventory/dto/request/UpdateUserPermissionsRequest.java
Comment (VladoKat) [inline src/main/java/jewellery/inventory/dto/request/UpdateUserPermissionsRequest.java:1]: Let's name this UpdateUserInOrganizationRequest because UpdateUserPermission is a bit deceptive - it sounds like updating general user authorization permissions

File: src/main/java/jewellery/inventory/exception/organization/MissingOrganizationPermissionException.java [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+public class MissingOrganizationPermissionException extends RuntimeException {
+  public MissingOrganizationPermissionException(
+      UUID userId, UUID organizationId, OrganizationPermission permission) {
+    super(
+        "The User with id "
+            + userId
+            + " not have "
+            + permission.toString()
+            + " permission for Organization with id "
+            + organizationId);
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/exception/organization/MissingOrganizationPermissionException.java]:
```suggestion
        "User with id "
            + userId
            + " does not have "
            + permission.toString()
            + " permission for Organization with id "
            + organizationId);
```

File: src/main/java/jewellery/inventory/mapper/OrganizationMapper.java [hunk 16]
```diff
@@ -26,4 +33,51 @@ public OrganizationResponseDto toResponse(Organization organization) {
     organizationResponseDto.setNote(organization.getNote());
     return organizationResponseDto;
   }
+
+  private List<UserInOrganizationResponseDto> toUserInOrganizationResponseDtoResponse(
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/mapper/OrganizationMapper.java]: I would argue we should just call all the methods in the mapper `map` but for now let's at least remove the redundant Reponse. 
Edit: This is also a list... But the suggestion looks bad. I am leaning towards `map` more and more 🤭 
```suggestion
  private List<UserInOrganizationResponseDto> toUserInOrganizationListResponseDto(
```

File: src/main/java/jewellery/inventory/mapper/OrganizationMapper.java [hunk 17]
```diff
@@ -26,4 +33,51 @@ public OrganizationResponseDto toResponse(Organization organization) {
...
+              UserInOrganizationResponseDto userResponseDto = new UserInOrganizationResponseDto();
+              userResponseDto.setUserId(userInOrg.getUser().getId());
+              userResponseDto.setOrganizationPermissions(userInOrg.getOrganizationPermission());
+              userResponseDtoList.add(userResponseDto);
+            });
+
+    return userResponseDtoList;
+  }
+
+  private UserInOrganizationResponseDto toUserInOrganizationResponseDtoResponse(
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/mapper/OrganizationMapper.java]: same here
```suggestion
  private UserInOrganizationResponseDto toUserInOrganizationResponseDto(
```

File: src/main/java/jewellery/inventory/mapper/OrganizationMapper.java [hunk 18]
```diff
@@ -26,4 +33,51 @@ public OrganizationResponseDto toResponse(Organization organization) {
...
+  }
+
+    public OrganizationMembersResponseDto toOrganizationMembersResponseDto(Organization organization) {
+        OrganizationMembersResponseDto membersResponseDto = new OrganizationMembersResponseDto();
+        membersResponseDto.setMembers(toUserInOrganizationResponseDtoResponse(organization));
+        membersResponseDto.setOrganization(toResponse(organization));
+        return membersResponseDto;
+    }
+
+    public OrganizationSingleMemberResponseDto toOrganizationSingleMemberResponseDto(UUID userId,Organization organization){
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/mapper/OrganizationMapper.java]: This doesn't look formatted. Please, format it

File: src/main/java/jewellery/inventory/service/UserInOrganizationService.java [hunk 19]
```diff
@@ -0,0 +1,191 @@
...
+@Service
+@AllArgsConstructor
+public class UserInOrganizationService implements EntityFetcher {
+  private static final Logger logger = LogManager.getLogger(UserInOrganizationService.class);
+  private final OrganizationRepository organizationRepository;
+  private final OrganizationMapper organizationMapper;
+  private final AuthService authService;
+  private final UserService userService;
+  private final UserInOrganizationRepository userInOrganizationRepository;
+  private final OrganizationService organizationService;
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/service/UserInOrganizationService.java:31]: We should have only `organizationService` here. Let's not use `organizationRepository` directly

File: src/main/java/jewellery/inventory/service/UserInOrganizationService.java [hunk 20]
```diff
@@ -0,0 +1,191 @@
...
+
+    UserInOrganization userInOrganization =
+        userInOrganizationRepository
+            .findByUserIdAndOrganizationId(userId, organizationId)
+            .orElseThrow(() -> new UserIsNotPartOfOrganizationException(userId, organizationId));
+
+    Organization organization = userInOrganization.getOrganization();
+    User userForUpdate = userInOrganization.getUser();
+
+    validateCurrentUserPermission(organization, OrganizationPermission.MANAGE_USERS);
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/service/UserInOrganizationService.java]: we should do the validation before fetching data from the db.
```suggestion
    validateCurrentUserPermission(organization, OrganizationPermission.MANAGE_USERS);
    UserInOrganization userInOrganization =
        userInOrganizationRepository
            .findByUserIdAndOrganizationId(userId, organizationId)
            .orElseThrow(() -> new UserIsNotPartOfOrganizationException(userId, organizationId));

    Organization organization = userInOrganization.getOrganization();
    User userForUpdate = userInOrganization.getUser();
```

File: src/main/java/jewellery/inventory/service/UserInOrganizationService.java [hunk 21]
```diff
@@ -0,0 +1,191 @@
...
+        organizationId,
+        userId);
+    return organizationMapper.toOrganizationSingleMemberResponseDto(userId, organization);
+  }
+
+  @LogCreateEvent(eventType = EventType.ORGANIZATION_USER_CREATE)
+  public OrganizationSingleMemberResponseDto addUserInOrganization(
+      UUID organizationId, UserInOrganizationRequestDto userInOrganizationRequestDto) {
+    Organization organization = organizationService.getOrganization(organizationId);
+    validateCurrentUserPermission(organization, OrganizationPermission.MANAGE_USERS);
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/service/UserInOrganizationService.java]: same here with the validation maybe
```suggestion
    validateCurrentUserPermission(organization, OrganizationPermission.MANAGE_USERS);
    Organization organization = organizationService.getOrganization(organizationId);
```

File: src/main/java/jewellery/inventory/service/UserInOrganizationService.java [hunk 22]
```diff
@@ -0,0 +1,191 @@
...
+  }
+
+  @LogUpdateEvent(eventType = EventType.ORGANIZATION_USER_UPDATE)
+  public OrganizationSingleMemberResponseDto updateUserPermissionsInOrganization(
+      UUID userId, UUID organizationId, List<OrganizationPermission> organizationPermissionList) {
+
+    UserInOrganization userInOrganization =
+        userInOrganizationRepository
+            .findByUserIdAndOrganizationId(userId, organizationId)
+            .orElseThrow(() -> new UserIsNotPartOfOrganizationException(userId, organizationId));
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/service/UserInOrganizationService.java]: Let's extract this into a method as it is used in the fetcher as well

File: src/test/java/jewellery/inventory/helper/UserTestHelper.java
Comment (VladoKat) [inline src/test/java/jewellery/inventory/helper/UserTestHelper.java:1]: Format the file

File: src/test/java/jewellery/inventory/integration/AuthIntegrationTest.java [hunk 23]
```diff
@@ -35,6 +33,7 @@ class AuthIntegrationTest extends AuthenticatedIntegrationTestBase {
   @BeforeEach
   void setUp() {
     this.setup();
+    userRepository.deleteAll();
```
Comment (VladoKat) [inline src/test/java/jewellery/inventory/integration/AuthIntegrationTest.java]: We have this method in AuthenticatedIntegrationTestBase::setup. Why do we need to delete the users again here?

File: src/test/java/jewellery/inventory/integration/AuthenticatedIntegrationTestBase.java [hunk 24]
```diff
@@ -66,9 +66,7 @@ void setup() {
     User adminUser = createTestUserWithId();
     setupMockSecurityContext(adminUser);
     setupTestRestTemplateWithAuthHeaders();
-    User adminDifferent = createUserInDatabase(UserTestHelper.createDifferentTestUserRequest());
-    setupMockSecurityContext(adminDifferent);
-    setupTestRestTemplateWithAuthHeaders();
+    adminUser.setId(createUserInDatabase(UserTestHelper.getTestUserRequest(adminUser)).getId());
```
Comment (VladoKat) [inline src/test/java/jewellery/inventory/integration/AuthenticatedIntegrationTestBase.java:68]: I guess that's why... I think we should consider putting the authenticated user as a different (adminUser) than the testUser we use to test other user related tests.

File: src/test/java/jewellery/inventory/integration/OrganizationCrudIntegrationTest.java [hunk 25]
```diff
@@ -28,18 +32,30 @@ private String getOrganizationByIdUrl(UUID id) {
     return "/organizations/" + id;
   }
 
+  private String getOrganizationUsersUrl(UUID organization, UUID user) {
+    return "/organizations/" + organization + "/users/" + user;
+  }
```
Comment (VladoKat) [inline src/test/java/jewellery/inventory/integration/OrganizationCrudIntegrationTest.java]:
```suggestion
  private String getOrganizationUsersUrl(UUID organizationId, UUID userId) {
    return "/organizations/" + organizationId + "/users/" + userId;
  }
```

File: src/test/java/jewellery/inventory/integration/OrganizationCrudIntegrationTest.java [hunk 26]
```diff
@@ -28,18 +32,30 @@ private String getOrganizationByIdUrl(UUID id) {
...
+
+  private String getOrganizationUsersUrl(UUID organization) {
+    return "/organizations/" + organization + "/users";
+  }
```
Comment (VladoKat) [inline src/test/java/jewellery/inventory/integration/OrganizationCrudIntegrationTest.java]:
```suggestion
  private String getOrganizationUsersUrl(UUID organizationId) {
    return "/organizations/" + organizationId + "/users";
  }
```

File: src/test/java/jewellery/inventory/integration/OrganizationCrudIntegrationTest.java [hunk 27]
```diff
@@ -69,13 +85,100 @@ void getOrganizationByIdSuccessfully() {
...
+            userInOrganizationRequestDto,
+            OrganizationSingleMemberResponseDto.class);
+
+    assertEquals(HttpStatus.CREATED, response.getStatusCode());
+    assertNotNull(response.getBody());
+
+    Map<String, Object> expectedEventPayload =
+        getCreateOrDeleteEventPayload(response.getBody(), objectMapper);
+
+    systemEventTestHelper.assertEventWasLogged(ORGANIZATION_USER_CREATE, expectedEventPayload);
```
Comment (VladoKat) [inline src/test/java/jewellery/inventory/integration/OrganizationCrudIntegrationTest.java:181]: We should add similar assertions for the create and delete tests as well

File: src/test/java/jewellery/inventory/integration/ResourceInUserCrudIntegrationTest.java [hunk 28]
```diff
@@ -186,7 +186,7 @@ void getAllUsersOwningResourceSuccessfully() {
 
     assertEquals(HttpStatus.OK, response.getStatusCode());
     assertNotNull(response.getBody());
-    assertEquals(2, response.getBody().getUsersAndQuantities().size());
+    assertEquals(1, response.getBody().getUsersAndQuantities().size());
```
Comment (VladoKat) [inline src/test/java/jewellery/inventory/integration/ResourceInUserCrudIntegrationTest.java]: Why did this change?

File: src/test/java/jewellery/inventory/integration/UserCrudIntegrationTest.java [hunk 29]
```diff
@@ -35,7 +33,7 @@ private String getBaseUserUrl() {
   @Test
   void createUserSuccessfully() throws Exception {
 
-    UserRequestDto userRequest = createTestUserRequest();
+    UserRequestDto userRequest = createDifferentTestUserRequest();
```
Comment (VladoKat) [inline src/test/java/jewellery/inventory/integration/UserCrudIntegrationTest.java]: I think all these tests shouldn't be changed if we put a different admin user as a currently authenticated user

**Review by VladoKat — CHANGES_REQUESTED** [review]
Things generally look well. There are some questionable decisions done for the tests that I think we can improve on. Let's discuss.

File: src/main/java/jewellery/inventory/mapper/OrganizationMapper.java [hunk 30]
```diff
@@ -26,4 +33,54 @@ public OrganizationResponseDto toResponse(Organization organization) {
...
+              UserInOrganizationResponseDto userResponseDto = new UserInOrganizationResponseDto();
+              userResponseDto.setUserId(userInOrg.getUser().getId());
+              userResponseDto.setOrganizationPermissions(userInOrg.getOrganizationPermission());
+              userResponseDtoList.add(userResponseDto);
+            });
+
+    return userResponseDtoList;
+  }
+
+  private UserInOrganizationResponseDto toUserInOrganizationListResponseDto(
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/mapper/OrganizationMapper.java]: This is not list
```suggestion
  private UserInOrganizationResponseDto toUserInOrganizationResponseDto(
```

File: src/main/java/jewellery/inventory/mapper/OrganizationMapper.java [hunk 31]
```diff
@@ -26,4 +33,54 @@ public OrganizationResponseDto toResponse(Organization organization) {
...
+  }
+
+  public OrganizationSingleMemberResponseDto toOrganizationSingleMemberResponseDto(
+      UUID userId, Organization organization) {
+    OrganizationSingleMemberResponseDto memberResponseDto =
+        new OrganizationSingleMemberResponseDto();
+    memberResponseDto.setMember(toUserInOrganizationListResponseDto(organization, userId));
+    memberResponseDto.setOrganization(toResponse(organization));
+    return memberResponseDto;
+  }
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/mapper/OrganizationMapper.java]: Let's move the public methods on top of the private ones

File: src/main/java/jewellery/inventory/service/OrganizationService.java [hunk 32]
```diff
@@ -58,9 +58,21 @@ private List<Organization> getAll() {
...
 
-  private Organization getOrganization(UUID id) {
+  public Organization getOrganization(UUID id) {
     return organizationRepository
         .findById(id)
         .orElseThrow(() -> new OrganizationNotFoundException(id));
   }
+  public void saveOrganization(Organization organization){
+    organizationRepository.save(organization);
+  }
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/service/OrganizationService.java]: same with these, move them before the private methods in the file

File: src/main/java/jewellery/inventory/service/OrganizationService.java [hunk 33]
```diff
@@ -58,9 +58,21 @@ private List<Organization> getAll() {
...
+
+  @Override
+  public Object fetchEntity(Object... ids) {
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/service/OrganizationService.java:71]: This can stay on the bottom as it is for the interface implementation

File: src/main/java/jewellery/inventory/service/UserInOrganizationService.java [hunk 34]
```diff
@@ -0,0 +1,198 @@
...
+
+  private void changeUserPermissionInOrganization(
+      Organization organization,
+      User userForUpdate,
+      List<OrganizationPermission> organizationPermissionList) {
+
+    organization.getUsersInOrganization().stream()
+        .filter(userInOrg -> userInOrg.getUser().equals(userForUpdate))
+        .findFirst()
+        .ifPresent(
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/service/UserInOrganizationService.java]: Are we certain here we are finding a user here every time? If we don't find a user this method will just pass without any errors. Should we try to protect ourselves by throwing and exception if it's not present?

File: src/test/java/jewellery/inventory/integration/UserCrudIntegrationTest.java
Comment (VladoKat) [inline src/test/java/jewellery/inventory/integration/UserCrudIntegrationTest.java:1]: This seems like an obsolete change.

File: src/test/java/jewellery/inventory/unit/service/UserInOrganizationServiceTest.java [hunk 35]
```diff
@@ -0,0 +1,127 @@
...
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class UserInOrganizationServiceTest {
+  @InjectMocks private UserInOrganizationService userInOrganizationService;
+  @Mock private OrganizationService organizationService;
+  @Mock private OrganizationRepository organizationRepository;
```
Comment (VladoKat) [inline src/test/java/jewellery/inventory/unit/service/UserInOrganizationServiceTest.java]: OrganizationRepository is not used. Please, remove it :)

**Review by VladoKat — CHANGES_REQUESTED** [review]

File: src/main/java/jewellery/inventory/mapper/OrganizationMapper.java [hunk 36]
```diff
@@ -26,4 +31,48 @@ public OrganizationResponseDto toResponse(Organization organization) {
...
+              UserInOrganizationResponseDto userResponseDto = new UserInOrganizationResponseDto();
+              userResponseDto.setUserId(userInOrg.getUser().getId());
+              userResponseDto.setOrganizationPermissions(userInOrg.getOrganizationPermission());
+              userResponseDtoList.add(userResponseDto);
+            });
+
+    return userResponseDtoList;
+  }
+
+  private UserInOrganizationResponseDto toListUserInOrganizationResponseDto(
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/mapper/OrganizationMapper.java]: Both this method and the one on top are named the same. When you refactor one, it refactors the other as well. In this case you should unselect the other method when refactoring
Screenshot 2024-03-08 at 10 29 58

File: src/test/java/jewellery/inventory/integration/AuthIntegrationTest.java
Comment (VladoKat) [inline src/test/java/jewellery/inventory/integration/AuthIntegrationTest.java:1]: this is still not rolled back to the test version

**Review by VladoKat — APPROVED** [review]
LGTM!
